fuzz/fuzz_packfile_raw.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * libgit2 raw packfile fuzz target.
+ *
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include <openssl/sha.h>
+
+#include "git2.h"
+
+#define REPOSITORY_PATH "fuzz.git"
+
+static git_repository *repo = NULL;
+static git_odb *odb = NULL;
+
+int LLVMFuzzerInitialize(__attribute__((unused)) int *argc,
+			 __attribute__((unused)) char ***argv)
+{
+	if (git_libgit2_init() < 0) {
+		fprintf(stderr, "Failed to initialize libgit2\n");
+		abort();
+	}
+	if (git_repository_init(&repo, REPOSITORY_PATH, true) < 0) {
+		fprintf(stderr, "Failed to initialize the repository\n");
+		abort();
+	}
+	if (git_repository_odb(&odb, repo) < 0) {
+		fprintf(stderr, "Failed to create the odb\n");
+		abort();
+	}
+	return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+	if (size == 0)
+		return 0;
+
+	if (!repo || !odb) {
+		fprintf(stderr, "Global state not initialized\n");
+		abort();
+	}
+
+	git_indexer *indexer;
+	git_transfer_progress stats = {0, 0};
```

Comment:
We are using C90 only, so please no variable declarations after statements.

Reply:
Done (switching my brain off of C++ is hard)